Dockerfile.rocm.ubi

```diff
@@ -3,133 +3,107 @@ ARG BASE_UBI_IMAGE_TAG=9.4
 
 RUN microdnf -y update && \
-    microdnf -y install rocm hipcc git which && \
+    microdnf -y install \
+        rocm-hip-libraries rocm-hip-runtime \
+        miopen-hip && \
     microdnf clean all
 
-WORKDIR /workspace
 
-##################################################################################################
+FROM rocm_base as rocm_devel
+ARG PYTORCH_ROCM_ARCH
 
-FROM rocm-base as python-install
-ARG PYTHON_VERSION
+ENV CCACHE_DIR=/root/.cache/ccache
 
-ENV VIRTUAL_ENV=/opt/vllm
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
-RUN microdnf install -y --setopt=install_weak_deps=0 --nodocs \
-    python${PYTHON_VERSION}-devel python${PYTHON_VERSION}-pip python${PYTHON_VERSION}-wheel && \
-    python${PYTHON_VERSION} -m venv $VIRTUAL_ENV --system-site-packages && \
-    $VIRTUAL_ENV/bin/pip install --no-cache -U pip wheel && \
+RUN rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    rpm -ql epel-release && \
+    microdnf -y update && \
+    microdnf -y install \
+        ccache \
+        git \
+        rocm \
+        hipcc \
+        wget \
+        which && \
     microdnf clean all
 
-##################################################################################################
-
-FROM python-install as python-rocm-base
-
-# install common dependencies
-RUN --mount=type=cache,target=/root/.cache/pip \
-    --mount=type=bind,source=requirements-common.txt,target=requirements-common.txt,readonly \
-    --mount=type=bind,source=requirements-rocm.txt,target=requirements-rocm.txt,readonly \
-    pip install -r requirements-rocm.txt
-
-##################################################################################################
+WORKDIR /workspace
 
-FROM python-rocm-base as base
-
-# Set the application mount point
-ARG APP_MOUNT=/vllm-workspace
-WORKDIR ${APP_MOUNT}
-
-# Upgrade pip and remove unnecessary packages
-RUN python3 -m pip install --upgrade --no-cache-dir pip && \
-    microdnf -y remove sccache || true && \
-    python3 -m pip uninstall -y sccache || true && \
-    rm -f "$(which sccache)" && \
-    microdnf clean all && \
-    rm -rf /var/cache/yum /var/cache/dnf
-
-# Install torch == 2.5.0 on ROCm
-RUN case "$(ls /opt | grep -Po 'rocm-[0-9]\.[0-9]')" in \
-    *"rocm-6.1"*) \
-    python3 -m pip uninstall -y torch torchvision \
-    && python3 -m pip install --no-cache-dir --pre \
-    torch==2.5.0.dev20240726 \
-    torchvision==0.20.0.dev20240726 \
-    --index-url https://download.pytorch.org/whl/nightly/rocm6.1;; \
-    *) ;; esac
-
-# Set environment variables
 ENV LLVM_SYMBOLIZER_PATH=/opt/rocm/llvm/bin/llvm-symbolizer
 ENV PATH=$PATH:/opt/rocm/bin:/libtorch/bin
-ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/rocm/lib:/libtorch/lib
 ENV CPLUS_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:/libtorch/include:/libtorch/include/torch/csrc/api/include:/opt/rocm/include
-ENV PYTORCH_ROCM_ARCH="gfx908;gfx90a;gfx942;gfx1100"
-ENV CCACHE_DIR=/root/.cache/ccache
+ENV PYTORCH_ROCM_ARCH=${PYTORCH_ROCM_ARCH}
```

Comment:
isn't this var used by vllm only later on..?

Reply:
Yes, we can remove this.